syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "objects.proto";
import "task_status.proto";
import "tasks_common.proto";

package armonik.api.grpc.v1.tasks;

option csharp_namespace = "Armonik.Api.gRPC.V1.Tasks";

/**
 * The TaskService provides methods for listing and getting tasks.
 */
service Tasks {
  // Get all tasks with pagination
  rpc ListTasks(ListTasksRequest) returns (ListTasksResponse) {}

  // Get a task by id
  rpc GetTask(GetTaskRequest) returns (GetTaskResponse) {}
}